### ISSUE TITLE
Moving scratch note fails during renaming

### ISSUE THREAD

**Issue by hikchoi** [dendronhq/dendron#480]
**Describe the bug**
Using the Move Note command on a scratch note fails during renaming the note.

This happens when:
1. moving scratch note to a child of an existing note 
(e.g. `scratch.2021.02.12.124511` -> `root.move-test-scratch`)
2. renaming within the scratch note hierarchy 
(e.g. `scratch.2021.02.12.124511` -> `scratch.2021.02.12.135678`)
3. moving scratch note to a new hierarchy 
(e.g. `scratch.2021.02.12.124511` -> `move-test-scratch`)

This does not happen with notes that were in an existing hierarchy other than scratch

**To Reproduce**
1. Make a scratch note
2. Use move note command on it to move it to any hierarchy
3. This will fail with `TypeError: Cannot read property 'forEach' of undefined at DendronEngineClient.refreshNotesV2`
4. Scratch note is not moved.

**Expected behavior**
Note is moved to a new hierarchy and the original is deleted (renamed)

**Desktop (please complete the following information):**
 - OS: MacOS Big Sur 11.0.1
 - Version: 0.28, also happens when I pull the latest code from master and build. 

**Dendron Log file**
<details><summary>to existing hierarchy</summary>
<p>
# Plugin Logs

```
/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)\"}"}}
{"level":30,"time":1613101686482,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","msg":"exit"}
{"level":30,"time":1613101705354,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"DiagnosticsReportCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101705361,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"execute","opts":{}}
{"level":30,"time":1613101705400,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"DiagnosticsReportCommand:run","msg":"post-execute"}
{"level":30,"time":1613101718088,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101718094,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","opts":{"oldLoc":{"fname":"scratch.2021.02.12.124511","vault":{"fsPath":"vault"}},"newLoc":{"fname":"root.move-test-scratch","vault":{"fsPath":"vault","label":"vault"}},"closeAndOpenFile":true}}
{"level":50,"time":1613101718103,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand:run","err":{"type":"DendronError","message":"","stack":"Error: \n\tat MoveNoteCommand.run (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/base.js:51:47)\n\tat async /Users/hikchoi/dev/dendron/packages/plugin-core/out/workspace.js:344:17\n\tat async b._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)","status":"unknown","msg":"","payload":"{\"msg\":\"Cannot read property 'forEach' of undefined\",\"stack\":\"TypeError: Cannot read property 'forEach' of undefined\\n\\tat DendronEngineClient.refreshNotesV2 (/Users/hikchoi/dev/dendron/packages/engine-server/lib/engineClient.js:167:15)\\n\\tat DendronEngineClient.renameNote (/Users/hikchoi/dev/dendron/packages/engine-server/lib/engineClient.js:193:20)\\n\\tat async MoveNoteCommand.execute (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/MoveNoteCommand.js:75:26)\\n\\tat async MoveNoteCommand.run (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/base.js:43:30)\\n\\tat async /Users/hikchoi/dev/dendron/packages/plugin-core/out/workspace.js:344:17\\n\\tat async b._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)\"}"}}
{"level":30,"time":1613101721106,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","msg":"exit"}
{"level":30,"time":1613101722283,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"DiagnosticsReportCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101722283,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"execute","opts":{}}
```

---
# Server Logs

```
 200 1.040 ms - 14 - 364
{"level":30,"time":1613101683472,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"NoteController:rename","msg":"enter"}
{"level":30,"time":1613101683472,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"Store:renameNote","msg":"enter","opts":{"oldLoc":{"fname":"scratch.2021.02.12.124511","vault":{"fsPath":"vault"}},"newLoc":{"fname":"scratch.2021.02.12.190384","vault":{"fsPath":"vault","label":"vault"}},"closeAndOpenFile":true}}
{"level":30,"time":1613101683474,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"c8652a53-d38e-4667-ac65-d5d5b8d60d1f","title":"124511","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12.124511","updated":1613101608846,"created":1613101513079,"parent":"05d7be18-1686-4383-8927-7096a484a4ad","children":[],"body":"\nmove note test\n","data":{},"custom":{}},"opts":{"metaOnly":true}}
{"level":30,"time":1613101683474,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"c8652a53-d38e-4667-ac65-d5d5b8d60d1f","title":"124511","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12.124511","updated":1613101608846,"created":1613101513079,"parent":"05d7be18-1686-4383-8927-7096a484a4ad","children":[],"body":"\nmove note test\n","data":{},"custom":{}},"msg":"delete from parent"}
POST /api/note/rename 200 3.186 ms - 849 - 224
{"level":30,"time":1613101718098,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"NoteController:rename","msg":"enter"}
{"level":30,"time":1613101718099,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"Store:renameNote","msg":"enter","opts":{"oldLoc":{"fname":"scratch.2021.02.12.124511","vault":{"fsPath":"vault"}},"newLoc":{"fname":"root.move-test-scratch","vault":{"fsPath":"vault","label":"vault"}},"closeAndOpenFile":true}}
{"level":30,"time":1613101718100,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"c8652a53-d38e-4667-ac65-d5d5b8d60d1f","title":"124511","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12.124511","updated":1613101608846,"created":1613101513079,"parent":"05d7be18-1686-4383-8927-7096a484a4ad","children":[],"body":"\nmove note test\n","data":{},"custom":{}},"opts":{"metaOnly":true}}
{"level":30,"time":1613101718100,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"c8652a53-d38e-4667-ac65-d5d5b8d60d1f","title":"124511","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12.124511","updated":1613101608846,"created":1613101513079,"parent":"05d7be18-1686-4383-8927-7096a484a4ad","children":[],"body":"\nmove note test\n","data":{},"custom":{}},"msg":"delete from parent"}
POST /api/note/rename 200 2.567 ms - 849 - 221
# Dendron Confg
{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron"}}
```

</p>
</details>

<details><summary>within scratch hierarchy</summary>
<p>

# Plugin Logs
---

```
:30,"time":1613101628100,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1613101628102,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"GoDownCommand:run","msg":"post-execute"}
{"level":30,"time":1613101628103,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":3}
{"level":30,"time":1613101628103,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"scratch.2021.02.12.124511.","source":"updatePickerBehavior:normal","profile":3,"cancelled":false}
{"level":30,"time":1613101683462,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101683468,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","opts":{"oldLoc":{"fname":"scratch.2021.02.12.124511","vault":{"fsPath":"vault"}},"newLoc":{"fname":"scratch.2021.02.12.190384","vault":{"fsPath":"vault","label":"vault"}},"closeAndOpenFile":true}}
{"level":50,"time":1613101683478,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand:run","err":{"type":"DendronError","message":"","stack":"Error: \n\tat MoveNoteCommand.run (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/base.js:51:47)\n\tat async /Users/hikchoi/dev/dendron/packages/plugin-core/out/workspace.js:344:17\n\tat async b._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)","status":"unknown","msg":"","payload":"{\"msg\":\"Cannot read property 'forEach' of undefined\",\"stack\":\"TypeError: Cannot read property 'forEach' of undefined\\n\\tat DendronEngineClient.refreshNotesV2 (/Users/hikchoi/dev/dendron/packages/engine-server/lib/engineClient.js:167:15)\\n\\tat DendronEngineClient.renameNote (/Users/hikchoi/dev/dendron/packages/engine-server/lib/engineClient.js:193:20)\\n\\tat async MoveNoteCommand.execute (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/MoveNoteCommand.js:75:26)\\n\\tat async MoveNoteCommand.run (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/base.js:43:30)\\n\\tat async /Users/hikchoi/dev/dendron/packages/plugin-core/out/workspace.js:344:17\\n\\tat async b._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)\"}"}}
{"level":30,"time":1613101686482,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","msg":"exit"}
{"level":30,"time":1613101705354,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"DiagnosticsReportCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101705361,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"execute","opts":{}}
```

---
# Server Logs

```
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"642e4c63-b6e2-4545-a0ef-387844e7bc59","title":"2021","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021","updated":1613101513086,"created":1613101513086,"parent":"4328e48d-523b-40f4-9004-f54e5aa38dc6","children":[],"body":"","data":{},"stub":true},"msg":"delete from parent"}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"4328e48d-523b-40f4-9004-f54e5aa38dc6","title":"Scratch","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch","updated":1613101513086,"created":1613101513086,"parent":"root","children":[],"body":"","data":{},"stub":true},"opts":{"metaOnly":true}}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"4328e48d-523b-40f4-9004-f54e5aa38dc6","title":"Scratch","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch","updated":1613101513086,"created":1613101513086,"parent":"root","children":[],"body":"","data":{},"stub":true},"msg":"delete from parent"}
POST /api/note/rename 200 4.633 ms - 1062 - 208
POST /api/note/update 200 1.166 ms - 14 - 364
POST /api/note/update 200 1.482 ms - 14 - 7426
POST /api/note/update 200 0.893 ms - 14 - 7428
POST /api/note/update 200 1.040 ms - 14 - 364
{"level":30,"time":1613101683472,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"NoteController:rename","msg":"enter"}
{"level":30,"time":1613101683472,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"Store:renameNote","msg":"enter","opts":{"oldLoc":{"fname":"scratch.2021.02.12.124511","vault":{"fsPath":"vault"}},"newLoc":{"fname":"scratch.2021.02.12.190384","vault":{"fsPath":"vault","label":"vault"}},"closeAndOpenFile":true}}
{"level":30,"time":1613101683474,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"c8652a53-d38e-4667-ac65-d5d5b8d60d1f","title":"124511","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12.124511","updated":1613101608846,"created":1613101513079,"parent":"05d7be18-1686-4383-8927-7096a484a4ad","children":[],"body":"\nmove note test\n","data":{},"custom":{}},"opts":{"metaOnly":true}}
{"level":30,"time":1613101683474,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"c8652a53-d38e-4667-ac65-d5d5b8d60d1f","title":"124511","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12.124511","updated":1613101608846,"created":1613101513079,"parent":"05d7be18-1686-4383-8927-7096a484a4ad","children":[],"body":"\nmove note test\n","data":{},"custom":{}},"msg":"delete from parent"}
POST /api/note/rename 200 3.186 ms - 849 - 224
# Dendron Confg
{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron"}}
```

</p>
</details>

<details><summary>to a new hierarchy</summary>
<p>

# Plugin Logs

```
,"msg":"enter"}
{"level":30,"time":1613101517063,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"WorkspaceWatcher:onWillSaveTextDocument","match":["updated: 1613101513079"],"msg":"update activeText editor"}
{"level":30,"time":1613101517675,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"VaultWatcher:onDidChange","uri":{"$mid":1,"fsPath":"/Users/hikchoi/dendron-test/vault/scratch.2021.02.12.124511.md","path":"/Users/hikchoi/dendron-test/vault/scratch.2021.02.12.124511.md","scheme":"file"}}
{"level":30,"time":1613101517677,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"VaultWatcher:onDidChange","fname":"scratch.2021.02.12.124511","msg":"exit"}
{"level":30,"time":1613101536563,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101536569,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","opts":{"oldLoc":{"fname":"scratch.2021.02.12.124511","vault":{"fsPath":"vault"}},"newLoc":{"fname":"move-test","vault":{"fsPath":"vault","label":"vault"}},"closeAndOpenFile":true}}
{"level":50,"time":1613101536580,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand:run","err":{"type":"DendronError","message":"","stack":"Error: \n\tat MoveNoteCommand.run (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/base.js:51:47)\n\tat async /Users/hikchoi/dev/dendron/packages/plugin-core/out/workspace.js:344:17\n\tat async b._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)","status":"unknown","msg":"","payload":"{\"msg\":\"Cannot read property 'forEach' of undefined\",\"stack\":\"TypeError: Cannot read property 'forEach' of undefined\\n\\tat DendronEngineClient.refreshNotesV2 (/Users/hikchoi/dev/dendron/packages/engine-server/lib/engineClient.js:167:15)\\n\\tat DendronEngineClient.renameNote (/Users/hikchoi/dev/dendron/packages/engine-server/lib/engineClient.js:193:20)\\n\\tat async MoveNoteCommand.execute (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/MoveNoteCommand.js:75:26)\\n\\tat async MoveNoteCommand.run (/Users/hikchoi/dev/dendron/packages/plugin-core/out/commands/base.js:43:30)\\n\\tat async /Users/hikchoi/dev/dendron/packages/plugin-core/out/workspace.js:344:17\\n\\tat async b._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:86:88509)\"}"}}
{"level":30,"time":1613101539585,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"MoveNoteCommand","msg":"exit"}
{"level":30,"time":1613101545195,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"DiagnosticsReportCommand:run","msg":"pre-execute"}
{"level":30,"time":1613101545201,"pid":11665,"hostname":"HikBookPro.local","name":"dendron","ctx":"execute","opts":{}}
```

---
# Server Logs

```
ToDelete":{"id":"05d7be18-1686-4383-8927-7096a484a4ad","title":"12","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02.12","updated":1613101513086,"created":1613101513086,"parent":"b4359224-a656-4083-a6a2-a3f271e977b7","children":[],"body":"","data":{},"stub":true},"msg":"delete from parent"}
{"level":30,"time":1613101536577,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"b4359224-a656-4083-a6a2-a3f271e977b7","title":"02","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02","updated":1613101513086,"created":1613101513086,"parent":"642e4c63-b6e2-4545-a0ef-387844e7bc59","children":[],"body":"","data":{},"stub":true},"opts":{"metaOnly":true}}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"b4359224-a656-4083-a6a2-a3f271e977b7","title":"02","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021.02","updated":1613101513086,"created":1613101513086,"parent":"642e4c63-b6e2-4545-a0ef-387844e7bc59","children":[],"body":"","data":{},"stub":true},"msg":"delete from parent"}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"642e4c63-b6e2-4545-a0ef-387844e7bc59","title":"2021","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021","updated":1613101513086,"created":1613101513086,"parent":"4328e48d-523b-40f4-9004-f54e5aa38dc6","children":[],"body":"","data":{},"stub":true},"opts":{"metaOnly":true}}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"642e4c63-b6e2-4545-a0ef-387844e7bc59","title":"2021","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch.2021","updated":1613101513086,"created":1613101513086,"parent":"4328e48d-523b-40f4-9004-f54e5aa38dc6","children":[],"body":"","data":{},"stub":true},"msg":"delete from parent"}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"4328e48d-523b-40f4-9004-f54e5aa38dc6","title":"Scratch","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch","updated":1613101513086,"created":1613101513086,"parent":"root","children":[],"body":"","data":{},"stub":true},"opts":{"metaOnly":true}}
{"level":30,"time":1613101536578,"pid":11665,"hostname":"HikBookPro.local","name":"dendron.server","ctx":"deleteNote","noteToDelete":{"id":"4328e48d-523b-40f4-9004-f54e5aa38dc6","title":"Scratch","vault":{"fsPath":"vault"},"type":"note","desc":"","links":[],"fname":"scratch","updated":1613101513086,"created":1613101513086,"parent":"root","children":[],"body":"","data":{},"stub":true},"msg":"delete from parent"}
POST /api/note/rename 200 4.633 ms - 1062 - 208
# Dendron Confg
{"version":1,"vaults":[{"fsPath":"vault"}],"useFMTitle":true,"site":{"copyAssets":true,"siteHierarchies":["root"],"siteRootDir":"docs","usePrettyRefs":true,"title":"Dendron"}}
```

</p>
</details>

**Comment by kevins8** [comment]
Some other users have reported this as well. Will be looking into it.
Thanks for the detailed logs!

**Comment by kevinslin** [comment]
This should be fixed in 0.29 :)
